src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -233,6 +239,32 @@ void FindBundleHeader()
             return headerOffset != 0;
         }
 
+        private static void CodeSign(string args)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                return;
+
+            const string codesign = @"/usr/bin/codesign";
+            if (!File.Exists(codesign))
+                return;
+
+            var p = new Process()
+            {
+                StartInfo = new ProcessStartInfo()
+                {
+                    Arguments = args,
+                    CreateNoWindow = true,
+                    ErrorDialog = false,
+                    FileName = codesign,
+                    UseShellExecute = false,
+                    WindowStyle = ProcessWindowStyle.Hidden,
+                    WorkingDirectory = Environment.CurrentDirectory
+                }
+            };
+
+            p.Start();
```

Comment:
Do you also need to wait for process to exit? And it would be nice to dispose the Process object once you are done with it. 
```
using (var p = Process.Start(...))
{
    p.WaitForExit();
}
```

Reply:
Also do you need to check the exit code?

Reply:
Not sure there's anything we can do if this fails.

Reply:
> Not sure there's anything we can do if this fails.

Print the error and fail the build step so that it does not fail silently?